internal/server.go

```diff
@@ -50,22 +59,33 @@ func (s *Server) buildRoutes() {
 }
 
 func (s *Server) RootHandler(w http.ResponseWriter, r *http.Request) {
+	logger := log.WithFields(logrus.Fields{
+		"X-Forwarded-Method": r.Header.Get("X-Forwarded-Method"),
+		"X-Forwarded-Proto":  r.Header.Get("X-Forwarded-Proto"),
+		"X-Forwarded-Host":   r.Header.Get("X-Forwarded-Host"),
+		"X-Forwarded-Prefix": r.Header.Get("X-Forwarded-Prefix"),
+		"X-Forwarded-Uri":    r.Header.Get("X-Forwarded-Uri"),
+	})
+
 	// Modify request
 	r.Method = r.Header.Get("X-Forwarded-Method")
 	r.Host = r.Header.Get("X-Forwarded-Host")
 	r.URL, _ = url.Parse(GetUriPath(r))
 
+	if config.PassAuthorization {
+		logger.Debug("pass_authorization enabled, creating token session")
+		if err := s.createTokenSession(w, r); err != nil {
```

Comment:
Is creating the session at the root handler necessary? Couldn't the `saveToken` create it if not found, and the `getTokenFromSession` handle appropriately if it wasn't there?

Reply:
It's not, now that passAuthorization is a configuration option. I originally tried to implement it as a query parameter, but traefik was filtering it out. So, good catch :)